ansible/ncn_nodes.yml

```diff
@@ -65,6 +65,46 @@
       when: ansible_distribution_file_variety == "SUSE"
     - role: csm.ncn.sat
 
+# Install SBPS iscsi target packages on iscsi targets(worker nodes)
+- hosts: Iscsi_Targets
+  gather_facts: yes
+  any_errors_fatal: true
+  remote_user: root
+  vars_files:
+    - vars/csm_repos.yml
+    - vars/csm_packages.yml
+  roles:
+    - role: csm.packages
+      vars:
+         packages: "{{ python-rtslib-fb-common + }targetcli-fb-common + python3-targetcli-fb}"
+      when: ansible_distribution_file_variety == "SUSE"
+
+# Config SBPS iscsi targets
+- hosts: Iscsi_Targets
+  gather_facts: yes
+  any_errors_fatal: true
+  remote_user: root
+  roles:
+    - role: csm.sbps.lio_config
+
+# Config SBPS DNS A and SRV records
+- hosts: Iscsi_Targets
+  gather_facts: yes
+    #delegate_to: localost
+  any_errors_fatal: true
+  remote_user: root
+  roles:
+    - role: csm.sbps.dns_srv_records
+
+# Mount s3 bucket 'boot-images' using s3fs read-only policy
+# for SBPS agent
+- hosts: Iscsi_Targets
+  gather_facts: yes
+  any_errors_fatal: true
+  remote_user: root
+  roles:
+    - role: csm.sbps.mount_s3_images
```

Comment:
Is there a reason you have these broken out into different blocks, when they all run on the same nodes?

```suggestion
# Install and configure SBPS on iSCSI targets (worker NCNs)
- hosts: Iscsi_Targets
  gather_facts: yes
  any_errors_fatal: true
  remote_user: root
  vars_files:
    - vars/csm_repos.yml
    - vars/csm_packages.yml
  roles:
    # Install SBPS iSCSI target packages
    - role: csm.packages
      vars:
        packages: "{{ python-rtslib-fb-common + }targetcli-fb-common + python3-targetcli-fb}"
      when: ansible_distribution_file_variety == "SUSE"
    # Configure SBPS
    - role: csm.sbps.lio_config
    # Configure SBPS DNS A and SRV records
    - role: csm.sbps.dns_srv_records
    # Mount s3 bucket 'boot-images' using s3fs read-only policy for SBPS agent
    - role: csm.sbps.mount_s3_images
```

Reply:
It was just to verify individually initially for UT. Will club all of them together now. Thanks.